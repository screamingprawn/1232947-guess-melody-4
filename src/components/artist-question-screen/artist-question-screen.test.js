import React from 'react';
import ArtistQuestionScreen from './artist-question-screen.jsx';
import renderer from 'react-test-renderer';

const question = {
  type: `genre`,
  genre: `rock`,
  song: {
    artist: `Billy Bob`,
    src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`
  },
  variants: [{
    picture: `https://api.adorable.io/avatars/128/0`,
    artist: `Jim Bim`,
  }, {
    picture: `https://api.adorable.io/avatars/128/1`,
    artist: `Billy Bob`,
  }, {
    picture: `https://api.adorable.io/avatars/128/2`,
    artist: `Ada Lovelace`,
  }],
};

it(`ArtistQuestionScreen is rendered correctly`, () => {
  const tree = renderer.create(
      <ArtistQuestionScreen
        question={question}
        onAnswer={() => {}}
        renderPlayer={() => {}}/>, {
       createNodeMock: () => {
          return {};
        }
      }
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
