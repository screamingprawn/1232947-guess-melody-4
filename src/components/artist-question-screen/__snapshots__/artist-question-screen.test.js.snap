// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistQuestionScreen is rendered correctly 1`] = `
<section
  className="game__screen"
>
  <h2
    className="game__title"
  >
    Кто исполняет эту песню?
  </h2>
  <div
    className="game__track"
  >
    <div
      className="track"
    />
  </div>
  <form
    className="game__artist"
  >
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="variant-0"
        name="artistAnswer"
        onChange={[Function]}
        type="radio"
        value="variant-0"
      />
      <label
        className="artist__name"
        htmlFor="variant-0"
      >
        <img
          alt="Jim Bim"
          className="artist__picture"
          src="https://api.adorable.io/avatars/128/0"
        />
        Jim Bim
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="variant-1"
        name="artistAnswer"
        onChange={[Function]}
        type="radio"
        value="variant-1"
      />
      <label
        className="artist__name"
        htmlFor="variant-1"
      >
        <img
          alt="Billy Bob"
          className="artist__picture"
          src="https://api.adorable.io/avatars/128/1"
        />
        Billy Bob
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="variant-2"
        name="artistAnswer"
        onChange={[Function]}
        type="radio"
        value="variant-2"
      />
      <label
        className="artist__name"
        htmlFor="variant-2"
      >
        <img
          alt="Ada Lovelace"
          className="artist__picture"
          src="https://api.adorable.io/avatars/128/2"
        />
        Ada Lovelace
      </label>
    </div>
  </form>
</section>
`;
